Cooper/ClientApp/src.v2/app/services/authentification/authentification.service.ts

```diff
@@ -0,0 +1,118 @@
+import { Injectable } from '@angular/core';
+import { HttpClient, HttpHeaders } from '@angular/common/http';
+import { Router } from '@angular/router';
+import { AuthService, FacebookLoginProvider, GoogleLoginProvider } from 'ng-dynami-social-login';
+
+@Injectable({
+  providedIn: 'root'
+})
+export class AuthentificationService {
+
+  readonly registrationUrl;
+  readonly authUrl;
+  readonly socialProvider;
+
+  constructor(private router: Router, private http: HttpClient, private socialAuthService: AuthService) {
+    this.registrationUrl = '/registration';
+    this.authUrl = '/auth/login';
+    this.socialProvider = {
+      'google': GoogleLoginProvider.PROVIDER_ID,
+      'facebook': FacebookLoginProvider.PROVIDER_ID
+    };
+   }
+
+  public checkAuthentification(): void {
+    const token: string = localStorage.getItem('JwtCooper');
+    if (token) {
+      this.router.navigate(['/platform/profile']);
+    }
+  }
+
+  public signIn(credentials: string): void {
+    this.http.post(this.authUrl, credentials, {
+      headers: new HttpHeaders({
+        'Content-Type': 'application/json'
+      })
+    }).subscribe(response => {
+      this.loginOK((response as any).token);
```

Comment:
any

Reply:
loginOk -> onSuccesLogin/succesAuthHandler
BadLogin -> onFailedLogin/failedAuthHandler

it isn't clear responsibility of these functions

Reply:
Now BadLogin() looks like redirectToLoginPage()